llama_hub/file/mbox/base.py

```diff
@@ -109,5 +111,9 @@ def load_data(
         docs: List[Document] = []
         content = self.parse_file(file)
         for msg in content:
-            docs.append(Document(text=msg, extra_info=extra_info or {}))
+            d = Document(text=msg, extra_info=extra_info or {})
+            if self.id_fn:
+                 d.id_ = self.id_fn(msg)
```

Comment:
canonical way to set doc_id is `d.doc_id = ...`.

Reply:
Thanks fixed! I think I had this from [here](https://github.com/jerryjliu/llama_index/blob/main/docs/core_modules/data_modules/documents_and_nodes/usage_documents.md?plain=1#L85)